File: src/main/java/org/cubexmc/provider/BukkitPermissionProvider.java
package org.cubexmc.provider;

import org.bukkit.entity.Player;

/**
 * Baseline permission provider for servers without an external permission
 * backend. Runtime permission attachments are managed by RuleGems managers.
 */
public class BukkitPermissionProvider implements PermissionProvider {

    @Override
    public void addPermission(Player player, String permission) {
        // RuleGems applies direct Bukkit PermissionAttachment nodes in managers.
    }

    @Override
    public void removePermission(Player player, String permission) {
        // RuleGems removes direct Bukkit PermissionAttachment nodes in managers.
    }

    @Override
    public void addGroup(Player player, String group) {
        // Bukkit has no native persistent group model.
    }

    @Override
    public void removeGroup(Player player, String group) {
        // Bukkit has no native persistent group model.
    }

    @Override
    public String getName() {
        return "Bukkit";
    }
}

File: src/main/java/org/cubexmc/provider/BukkitPermissionProvider.kt
package org.cubexmc.provider

import org.bukkit.entity.Player

/**
 * Baseline permission provider for servers without an external permission
 * backend. Runtime permission attachments are managed by RuleGems managers.
 */
open class BukkitPermissionProvider : PermissionProvider {
    override fun addPermission(player: Player, permission: String) {
        // RuleGems applies direct Bukkit PermissionAttachment nodes in managers.
    }

    override fun removePermission(player: Player, permission: String) {
        // RuleGems removes direct Bukkit PermissionAttachment nodes in managers.
    }

    override fun addGroup(player: Player, group: String) {
        // Bukkit has no native persistent group model.
    }

    override fun removeGroup(player: Player, group: String) {
        // Bukkit has no native persistent group model.
    }

    override fun getName(): String = "Bukkit"
}
